### PR TITLE
Error Trace Decorator

### PR DESCRIPTION
**Pull request by SparshaSaha**
### Description

This PR introduces a new tool named, `react-native-error-trace-decorator` which enables developers to symbolicate stack traces from react native apps which contain multiple bundles. (Currently experimental)

### Test plan

UTs have been written which has a 100% coverage across all files.
Tested locally against custom error stacks and sourcemaps

### CODE REVIEW

File: incubator/react-native-error-trace-decorator/src/extract_and_process_error_stack.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ * @param errorFilePath   The path to the error file
+ * @param configFilePath  The path to the config file
+ */
+export const extractAndSymbolicateErrorStack = (
+  errorFilePath: string,
+  configFilePath: string
+) => {
+  // Read config file as an object
+  const configFile = JSON.parse(
+    fse.readFileSync(configFilePath, "utf-8")
```
Comment (ravikrishnan1289) [inline incubator/react-native-error-trace-decorator/src/extract_and_process_error_stack.ts]: could we use the format parameterised or a constant? in line 32 its specified as "utf8"

File: incubator/react-native-error-trace-decorator/src/extract_and_process_error_stack.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        if (index === configFile.configs.length - 1) {
+          console.log(errorLine);
+        }
+        return true;
+      });
+    });
+  }
+};
+
+const checkIfConfigFileIsValidAndPopulateMap = (
```
Comment (ravikrishnan1289) [inline incubator/react-native-error-trace-decorator/src/extract_and_process_error_stack.ts]: let's rename the method to isConfigFileValid, the method name seems like handling multiple responsibilities

File: incubator/react-native-error-trace-decorator/src/extract_and_process_error_stack.ts (hunk 1 above)
Comment (SparshaSaha) [reply incubator/react-native-error-trace-decorator/src/extract_and_process_error_stack.ts]: I don't think we need a constant for that.  "utf-8" denotes the encodingin which the file must be read.

File: incubator/react-native-error-trace-decorator/src/extract_and_process_error_stack.ts (hunk 2 above)
Comment (SparshaSaha) [reply incubator/react-native-error-trace-decorator/src/extract_and_process_error_stack.ts]: Renamed this

File: incubator/react-native-error-trace-decorator/src/check_arguments_validity.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    printHelp();
+    return false;
+  }
+
+  // If config file does not exist, print out the appropriate error message
+  if (!fse.existsSync(args.configFile)) {
+    error(
+      `Config file does not exist in ${args.configFile}. Please pass a valid path`
+    );
+    return false;
```
Comment (prabhuvi) [inline incubator/react-native-error-trace-decorator/src/check_arguments_validity.ts:49]: Should help be printed here? Does is show what a valid path looks like? An example maybe

**Review by prabhuvi — APPROVED** [review]

File: incubator/react-native-error-trace-decorator/src/check_arguments_validity.ts (hunk 3 above)
Comment (SparshaSaha) [reply incubator/react-native-error-trace-decorator/src/check_arguments_validity.ts:49]: This part checks if the path provided for config file actually contains a file. I think we can skip printing help here

**Review by ravikrishnan1289 — APPROVED** [review]

**Pull request merged**